src/test/run-make/issue-24445/foo.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![crate_type = "staticlib"]
+
+struct Destroy;
+impl Drop for Destroy {
+    fn drop(&mut self) { println!("drop"); }
```

Comment:
Oh this was actually just here to help me debug it, it's triggered via the use of `X` instead of the contents of this destructor.